cdc/entry/mounter.go

```diff
@@ -219,29 +219,31 @@ func (m *mounterImpl) unmarshalRowKVEntry(tableInfo *model.TableInfo, rawKey []b
 	}, nil
 }
 
-const (
-	ddlJobListKey         = "DDLJobList"
-	ddlAddIndexJobListKey = "DDLJobAddIdxList"
-)
-
-// UnmarshalDDL unmarshals the ddl job from RawKVEntry
-func UnmarshalDDL(raw *model.RawKVEntry) (*timodel.Job, error) {
-	if raw.OpType != model.OpTypePut || !bytes.HasPrefix(raw.Key, metaPrefix) {
-		return nil, nil
-	}
-	meta, err := decodeMetaKey(raw.Key)
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-	if meta.getType() != ListData {
-		return nil, nil
-	}
-	k := meta.(metaListData)
-	if k.key != ddlJobListKey && k.key != ddlAddIndexJobListKey {
-		return nil, nil
+// ParseDDLJob parses the job from the raw KV entry. id is the column id of `job_meta`.
+func ParseDDLJob(tblInfo *model.TableInfo, rawKV *model.RawKVEntry, id int64) (*timodel.Job, error) {
+	var v []byte
+	if bytes.HasPrefix(rawKV.Key, MetaPrefix) {
+		// old queue base job.
+		v = rawKV.Value
+	} else {
+		recordID, err := tablecodec.DecodeRowKey(rawKV.Key)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		row, err := decodeRow(rawKV.Value, recordID, tblInfo, time.UTC)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		datum := row[id]
+		v = datum.GetBytes()
 	}
```

Comment:
Please add more comment on this logic. Maybe you can give a reference to the changes that have happened in TiDB.